pkg/broker/ingress/multi_topic_decouple_sink.go

```diff
@@ -32,19 +38,23 @@ import (
 const projectEnvKey = "PROJECT_ID"
 
 // NewMultiTopicDecoupleSink creates a new multiTopicDecoupleSink.
-func NewMultiTopicDecoupleSink(ctx context.Context, brokerConfig config.ReadonlyTargets, client cev2.Client) *multiTopicDecoupleSink {
+func NewMultiTopicDecoupleSink(ctx context.Context, brokerConfig config.ReadonlyTargets, client *pubsub.Client) *multiTopicDecoupleSink {
 	return &multiTopicDecoupleSink{
 		logger:       logging.FromContext(ctx),
-		client:       client,
+		pubsub:       client,
 		brokerConfig: brokerConfig,
+		topics:       make(map[types.NamespacedName]*pubsub.Topic),
 	}
 }
 
 // multiTopicDecoupleSink implements DecoupleSink and routes events to pubsub topics corresponding
 // to the broker to which the events are sent.
 type multiTopicDecoupleSink struct {
-	// client talks to pubsub.
-	client cev2.Client
+	// pubsub talks to pubsub.
+	pubsub *pubsub.Client
+	// map from brokers to topics
+	topics    map[types.NamespacedName]*pubsub.Topic
```

Comment:
It seems that if a broker is deleted, its entry is not removed from this map. To do the removal, I think we need to watch for the configmap file.

Reply:
Yes, good catch, I've already created #1118 to track this. Note this is also an issue when using the pubsub transport as we do currently.

Reply:
Right, I believe the existing CE SDK client doesn't provide a way for the clean up. Looks like it should be implemented here: https://github.com/cloudevents/sdk-go/blob/master/protocol/pubsub/v2/protocol.go#L243

Reply:
Can you add a TODO here and link the issue?